mood-analysis.rb

```diff
@@ -28,5 +28,65 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+# puts analyze_mood(text[0])
+# puts analyze_mood(text[1])
+
+def string_analysis(array)
+  array.each do |day|
+    p array[0][0..4] + analyze_mood(day)
+  end
+end
+
+# string_analysis(text)
+# 3. Write a method called `happy_days` to determine how many logged entries it takes until there have been three :-) happy days.
+def happy_days(words)
+  count =[]
+  days = 0
+  words.each do |day|
+    moods = analyze_mood(day)[-3..-1]
+    count << moods.include?(":-)")
+      days += 1
+    if count.length == 3
+      puts "It takes #{days} entries for 3 happy days to occur"
+      break
+    end
+  end
+  puts "sorry, to many moods!"
+end
+
+# happy_days(text)
+
+# 4. Write a method called `overall_mood` to determine the most common mood across all logged entries.
+
+def overall_mood(words)
+  happy = []
+  sad = []
+  neutral = []
+
+  words.each do |day|
+    moods = analyze_mood(day)[-3..-1]
+    if moods.include?(":-)")
+      happy << moods
+    elsif moods.include?(":-(")
+      sad << moods
+    else
+      neutral << moods
+    end
+  end
+
+    p happy
+    p sad
+    p neutral
+
+      if happy.length > sad.length && neutral.length
+        puts "The most common mood is :-)"
+          return happy
+      elsif sad.length > happy.length && neutral.length
+        puts "The most common mood is :-("
+          return sad
+      else
+        puts "The most common mood is :-|"
+          return neutral
+      end
+end
```

Comment:
Also, again here I would have liked to see the smallest piece of information needed to be returned. In this case, just a single emoticon.